src/parseTools.js

```diff
@@ -1569,3 +1569,17 @@ function getPageSize() {
   return WASM ? WASM_PAGE_SIZE : ASMJS_PAGE_SIZE;
 }
 
+// Receives a function as text, and returns modified text.
+function modifyFunction(text, func) {
+  var match = text.match(/\s*function\s+([^(]*)?\s*\(([^)]*)\)/);
+  assert(match, 'could not match function ' + text + '.');
+  var name = match[1];
+  var args = match[2];
+  var rest = text.substr(match[0].length);
+  var bodyStart = rest.indexOf('{');
+  assert(bodyStart > 0);
+  var bodyEnd = rest.lastIndexOf('}');
+  assert(bodyEnd > 0);
+  return func(name, args, rest.substring(bodyStart + 1, bodyEnd));
```

Comment:
I'm having trouble seeing there `func` is defined.  Does it return a string?

Reply:
I added a comment to clarify this now - basically we do the parsing here, and pass the parsed name, arguments, and body to the passed-in function which returns whatever it wants. (In the actual uses, it returns the string, but other uses could do otherwise.)